1.x: Deprecate CompositeException constructor with message prefix

vanniktech opened the pull request:
Before this PR messagePrefix was never used :dancer: 

If you guys want I can revert the change on CompositeExceptionTest.java:177. I just thought that way it' be nicer.

About the message is there any reason for having a space after the dot?

akarnokd commented:
We don't really use that constructor or the message, so I'd prefer removing it altogether.

vanniktech commented:
Yeah I also realized that later since no test broke after that change. So do you want me to remove it?

akarnokd commented:
Yes, but don't rush it as there is another, much heavier PR #3759 that also changed this class.

vanniktech commented:
Alright just ping me once #3759 is in

vanniktech commented:
@akarnokd I updated the PR again and changed the commit message and the title. I hope the changes are okay now. If the removal of whitespace bugs you just let me know.

akarnokd commented:
Yes, please revert those space changes. (Are you using IntelliJ?)

vanniktech commented:
Those changes I did in Sublime and I set it up in a way that it should remove trailing whitespace ;)

Also updated the PR and removed the spaces

akarnokd commented:
Almost, but forgot that its a public API component thus can't just remove the constructors. Could you instead deprecate `CompositeException(String message, List<Throwable> errors)`?

vanniktech commented:
@akarnokd like this?

akarnokd commented:
Yes.

:+1:

artem-zinnatullin commented:
:+1:

stevegury commented:
:+1:
